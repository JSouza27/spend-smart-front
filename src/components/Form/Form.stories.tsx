import { ComponentMeta, ComponentStory } from '@storybook/react'
import { ThemeProvider } from 'styled-components';

import Form from '.'
import theme from '../../styles/theme';

export default {
  title: 'Form',
  component: Form
} as ComponentMeta<typeof Form>

export const Basic: ComponentStory<typeof Form> = () => (
  <ThemeProvider theme={theme}>
    <Form />
  </ThemeProvider>
)
