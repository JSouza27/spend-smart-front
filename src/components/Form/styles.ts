import styled from 'styled-components';

export const Wrapper = styled.form`
  align-items: center;
  display: flex;
  flex-direction: column;
  gap: 1.6rem;
  max-width: 40rem;
  width: 100%;
`;
