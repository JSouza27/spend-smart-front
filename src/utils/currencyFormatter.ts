export class CurrencyFormatter {
  private static language = 'pt-BR';
  private static currency = 'BRL';

  static formatter(num: number): string {
    return (num ?? 0).toLocaleString(this.language, {
      currency: this.currency
    });
  }

  static unformat(num: string): number {
    const nums = num.replace(/[^0-9]+/g, '');
    const i = nums.length - 2;

    return Number(`${nums.substring(0, i)}.${nums.substring(i)}`);
  }
}
