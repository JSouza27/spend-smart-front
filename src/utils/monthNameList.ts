export const monthNameList = [
  'janeiro',
  'fevereiro',
  'mar√ßo',
  'abril',
  'maio',
  'junho',
  'julho',
  'agosto',
  'setembro',
  'outubro',
  'novembro',
  'dezembro'
];
