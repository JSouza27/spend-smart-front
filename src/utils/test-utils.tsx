import { render } from '@testing-library/react';
import React from 'react';
import { ThemeProvider } from 'styled-components';

import theme from '../styles/theme';
import { FormProvider, useForm } from 'react-hook-form';

const AllTheProviders = ({ children }: any) => {
  const methods = useForm();

  return (
    <ThemeProvider theme={theme}>
      <FormProvider {...methods}>{children}</FormProvider>
    </ThemeProvider>
  );
};

const customRender = (
  ui: React.ReactElement<any, string | React.JSXElementConstructor<any>>,
  options?: any
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
