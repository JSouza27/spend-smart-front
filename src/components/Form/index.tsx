import React, { FormHTMLAttributes } from 'react';

import * as S from './styles';

export type FormProps = {
  children: React.ReactNode;
} & FormHTMLAttributes<HTMLFormElement>;

export default function Form({ children }: FormProps) {
  return <S.Wrapper>{children}</S.Wrapper>;
}
