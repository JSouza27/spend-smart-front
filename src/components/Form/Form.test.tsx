import { render, screen } from 'utils/test-utils';

import Form from './index';

function ElementForm() {
  return (
    <>
      <label>Label</label>
      <input type="text" />
    </>
  );
}
describe('<Form />', () => {
  it('should render the Form', () => {
    const { container } = render(
      <Form>
        <ElementForm />
      </Form>
    );
    screen.logTestingPlaygroundURL();
    const label = screen.getByText(/label/i);
    const input = screen.getByRole('textbox');

    expect(label).toBeInTheDocument();
    expect(input).toBeInTheDocument();
    expect(container.firstChild).toMatchSnapshot();
  });
});
