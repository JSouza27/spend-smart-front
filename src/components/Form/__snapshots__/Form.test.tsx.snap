// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render the Form 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1.6rem;
  max-width: 40rem;
  width: 100%;
}

<form
  class="c0"
>
  <label>
    Label
  </label>
  <input
    type="text"
  />
</form>
`;
